onnxruntime/core/providers/migraphx/migraphx_execution_provider_utils.h

```diff
@@ -250,4 +252,71 @@ inline std::string GetCachePath(const std::string& root, const std::string& name
   }
 }
 
+inline std::string GenerateGraphId(const GraphViewer& graph_viewer) {
+  HashValue model_hash;
+
+  // find the top level graph
+  const Graph* cur_graph = &graph_viewer.GetGraph();
+  while (cur_graph->IsSubgraph()) {
+    cur_graph = cur_graph->ParentGraph();
+  }
+
+  const Graph& main_graph = *cur_graph;
+  uint32_t hash[4] = {0, 0, 0, 0};
+
+  auto hash_str = [&hash](const std::string& str) {
+    MurmurHash3::x86_128(str.data(), gsl::narrow_cast<int32_t>(str.size()), hash[0], &hash);
+  };
+
+  // Use the model's file name instead of the entire path to avoid cache regeneration if a path changes
+  const fs::path path{main_graph.ModelPath()};
```

Comment:
That still doesn't stop the case between model.onnx being used for different models add something like the name of the first n_nodes instead as a fingerprint.

Reply:
That only generates the <graph_id> section for an MXR file name. The MXT file name format is `<migraphx_version>-<graph_id>-<token_hash>.mxr`. The `migraphx_version` and `graph_id` are constant for a session. The `<token_hash>` changes depending on the inputs and can only be determined in the `MIGraphXExecutionProvider::Compile` method (see the method implementation for details).